honeybadgermpc/avss_value_processor.py

```diff
@@ -149,6 +133,37 @@ def _process_acs_output(self, acs_outputs):
                 if j < len(self.inputs_per_dealer[i]):
                     future.set_result(self.inputs_per_dealer[i][j])
 
+        self._add_to_output_queue()
+
+    def _add_to_output_queue(self):
+        # TODO: This can definitelt be optimized for space.
+        # The first loop is probably not needed.
+
+        # Get all the values dealt by each dealer which have been agreed and not
+        # yet added to the output queue. i_th row represents the set of values
+        # dealt by the i_th dealer.
+        pending_values = [None]*self.n
+        pending_counts = [0]*self.n
+        for i in range(self.n):
+            s, e = self.next_idx_to_return_per_dealer[i], len(self.outputs_per_dealer[i])
+            assert e - s >= 0
+            pending_values[i] = list(self.outputs_per_dealer[i][s:])
+            pending_counts[i] = len(pending_values[i])
+
+        pending_counts.sort()
+        # The t_th index represents the maximum values which at least `n-t` nodes have
```

Comment:
Is the sort ascending or descending? The reason I'm asking.... we only want to need to ensure that at least `t+1` nodes have received the value. So it would be OK to take the `t`'th *largest* count, not just the `t` smallest

Reply:
Here's why I took the `t_th` smallest element and not the largest:
```
|  0  |  1  |  2  |  3  |
-------------------------
| 00  | 10  | 20  | 30  |  => Each row is a batch.
| 01  | 11  | 21  | 31  |
|     | 12  | 22  | 32  |
|     |     | 23  | 33  |
|     |     |     | 34  |
|     |     |     | 35  |

Counts => [2, 3, 4, 6]

Sorted in ascending order, t_th idx value = 3
Batch 1 => 00, 10, 20, 30
Batch 2 => 01, 11, 21, 31
Batch 3 => 12, 22, 32

As per my understanding, you want values from at least `n-t` nodes in one batch.

AVSS Value Proessor Output Order => 00, 10, 20, 30, 01, 11, 21, 31, 12, 22, 32

If you sort them in descending order and then take the t_th idx value = 4 then there will be an extra batch:
Batch 4 => 23, 33
```